src/v/storage/disk_log_impl.cc

```diff
@@ -334,15 +334,39 @@ ss::future<> disk_log_impl::do_compact(compaction_config cfg) {
       "[{}] applying 'compaction' log cleanup policy with config: {}",
       config().ntp(),
       cfg);
-    // find first not compacted segment
+
+    // create a logging predicate for offsets..
+    auto offsets_compactible = [&](segment& s) {
```

Comment:
do we only need `cfg` reference here ?

Reply:
Yep. Do you prefer explicit `&cfg`? I couldn't think of an example of where default capture by reference *from* a coroutine is bad. Obviously capturing a reference to a _non-coroutine_ automatic var is bad, but this is a coroutine. Maybe good hygiene to always be explicit? The lambda its self is not a coroutine.

Reply:
the code is safe. but generally the code base (the exception being tests) avoids capturing the entire context for hygiene reasons.

Reply:
I'm cool with either. I love concise things but yeah, in non-test code extra hygiene is good. I'll update this, thank you.

Reply:
I also like being explicit for captures, lifetimes and the cost of copies should be explicit.